src/repository/Repository.ts

```diff
@@ -345,5 +345,20 @@ export class Repository<Entity extends ObjectLiteral> {
     async clear(): Promise<void> {
         return this.manager.clear(this.metadata.target);
     }
-
+    
+    /**
+     * Reloads the supplied entity into the latest state of the record from the database
+     */
+    reload(entity: Entity): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            const newEntity = await this.findOneById(entity.id);
```

Comment:
Let's make this feature part of `@next` branch which will be released in a few weeks. In `@next` there is a method called `findOneOrFail` that you can use instead of this check

Reply:
Got it, will do